packages/contracts/contracts/protocol/incentives/IncentivesController2.sol

```diff
@@ -0,0 +1,273 @@
+                        _incentivizedAssets[asset].decimals
+                    );
+                unchecked {
+                    ++j;
+                }
+            }
+            unchecked {
+                ++i;
+            }
+        }
+
+        return (rewards, amounts);
+    }
+
+    /**
+     * @dev Claims all available rewards on the given incentivized assets
+     * @param incentivizedAssets The list of incentivized asset addresses
+     * @param to The address to send the claimed funds to
+     * @return rewards The list of possible reward addresses
+     * @return amounts The list of amounts of each reward claimed
+     *
+     */
+    function claimAllRewards(address[] calldata incentivizedAssets, address to)
+        external
+        override
+        returns (address[] memory, uint256[] memory)
+    {
+        address[] memory rewards = _allRewards;
+        uint256[] memory amounts = new uint256[](rewards.length);
+        uint256[] memory penaltyAmounts = new uint256[](rewards.length);
+
+        DistributionTypes.UserAssetState[] memory userState = _getUserState(incentivizedAssets, msg.sender);
+        _batchUpdate(msg.sender, userState);
+
+        uint256 veTotalSupply = IVeVmex(VE_VMEX).totalSupply();
+        uint256 veUserBalance = IVeVmex(VE_VMEX).balanceOf(msg.sender);
+
+        uint256 aTokenTotalSupply;
+        uint256 boostedAmount;
+        for (uint256 i; i < incentivizedAssets.length;) {
+            address asset = incentivizedAssets[i];
+            aTokenTotalSupply = _standardizeDecimals(IERC20(asset).totalSupply(), _incentivizedAssets[asset].decimals);
+            for (uint256 j; j < rewards.length;) {
+                uint256 amount = _incentivizedAssets[asset].rewardData[rewards[j]].users[msg.sender].accrued;
+                boostedAmount = _calculateBoostedClaimable(veTotalSupply, aTokenTotalSupply, veUserBalance, amount);
+                if (amount != 0) {
+                    amounts[j] += boostedAmount;
+                    _incentivizedAssets[asset].rewardData[rewards[j]].users[msg.sender].accrued = 0;
+                    // if boostedAmount == amount boost is 100% so penalty is 0
+                    if (boostedAmount != amount) {
+                        unchecked {
+                            // boostedAmount is always less than amount
+                            penaltyAmounts[j] = amount - boostedAmount;
+                        }
+                    }
+                }
+
+                unchecked {
+                    ++j;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        uint256 amountsLength = amounts.length;
+        for (uint256 i; i < amountsLength;) {
+            if (amounts[i] != 0) {
+                IERC20(rewards[i]).safeTransferFrom(REWARDS_VAULT, to, amounts[i]);
+                IERC20(rewards[i]).safeTransferFrom(REWARDS_VAULT, PENALTY_RECIEVER, penaltyAmounts[i]);
```

Comment:
Ok, but in that case DistributionManager becomes eligible only for dVMEX rewards. All other rewards will fail, because burn function call on dVMEX reward poll will fail